modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -0,0 +1,1017 @@
+
+    package = mkOption {
+      inherit visible;
+      type = with types; nullOr package;
+      default = pkgs.${defaultPackageName};
+      defaultText = literalExpression "pkgs.${packageName}";
+      example = literalExpression ''
+        pkgs.${packageName}.override {
+          # See nixpkgs' firefox/wrapper.nix to check which options you can use
+          nativeMessagingHosts = [
+            # Gnome shell native connector
+            pkgs.gnome-browser-connector
+            # Tridactyl native connector
+            pkgs.tridactyl-native
+          ];
+        }
+      '';
+      description = ''
+        The ${name} package to use. If state version ≥ 19.09 then
+        this should be a wrapped ${name} package. For earlier state
+        versions it should be an unwrapped ${name} package.
+        Set to `null` to disable installing ${name}.
+      '';
+    };
+
+    languagePacks = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = ''
+        The language packs to install. Available language codes can be found
+        on the releases page:
+        `https://releases.mozilla.org/pub/firefox/releases/''${version}/linux-x86_64/xpi/`,
+        replacing `''${version}` with the version of Firefox you have.
+      '';
+      example = [ "en-GB" "de" ];
+    };
+
+    name = mkOption {
+      internal = true;
+      type = types.str;
+      default = name;
+      example = "Firefox";
+      description = "The name of the browser.";
+    };
+
+    wrappedPackageName = mkOption {
+      internal = true;
+      type = with types; nullOr str;
+      default = wrappedPackageName;
+      description = "Name of the wrapped browser package.";
+    };
+
+    vendorPath = mkOption {
+      internal = true;
+      type = with types; nullOr str;
+      default = with platforms;
+        if isDarwin then
+          darwin.vendorPath or null
+        else
+          linux.vendorPath or null;
+      example = ".mozilla";
+      description =
+        "Directory containing the native messaging hosts directory.";
+    };
+
+    configPath = mkOption {
+      internal = true;
+      type = types.str;
+      default = with platforms;
+        if isDarwin then darwin.configPath else linux.configPath;
+      example = ".mozilla/firefox";
+      description = "Directory containing the ${name} configuration files.";
+    };
+
+    nativeMessagingHosts = optionalAttrs (cfg.vendorPath != null) (mkOption {
+      inherit visible;
+      type = types.listOf types.package;
+      default = [ ];
+      description = ''
+        Additional packages containing native messaging hosts that should be
+        made available to ${name} extensions.
+      '';
+    });
+
+    finalPackage = mkOption {
+      inherit visible;
+      type = with types; nullOr package;
+      readOnly = true;
+      description = "Resulting ${cfg.name} package.";
+    };
+
+    policies = optionalAttrs (unwrappedPackageName != null) (mkOption {
```

Comment:
@brckd shouldn't this be dependet on `wrappedPackageName`?

Reply:
Yes, you are absolutely right! This should use `isWrapped`, since that is also used by `wrapPackage` to decide whether the policies can be overridden.

I probably didn't notice this during testing, as the configs I tested it with had unwrapped packages.

Reply:
I will work on a fix once I'm back from vacation, but anyone else feel free to attempt it in the meantime!

Reply:
Created #5841.